packages/plugins/apollo-usage-report/src/index.ts

```diff
@@ -55,24 +59,87 @@ type ApolloUsageReportOptions = ApolloInlineTracePluginOptions & {
    * Client version to report to the usage reporting API
    */
   clientVersion?: StringFromRequestFn;
+  /**
+   * The version of the runtime (like 'node v23.7.0')
+   * @default empty string.
+   */
+  runtimeVersion?: string;
+  /**
+   * The hostname of the machine running this server
+   * @default $HOSTNAME environment variable
+   */
+  hostname?: string;
+  /**
+   * The OS identification string.
+   * The format is `${os.platform()}, ${os.type()}, ${os.release()}, ${os.arch()})`
+   * @default empty string
+   */
+  uname?: string;
+  /**
+   * The maximum estimated size of each traces in bytes. If the estimated size is higher than this threshold,
+   * the complete trace will not be sent and will be reduced to aggregated stats.
+   *
+   * Note: GraphOS only allow for traces of 10mb maximum
+   * @default 10 * 1024 * 1024 (10mb)
+   */
+  maxTraceSize?: number;
+  /**
+   * The maximum uncompressed size of a report in bytes.
+   * The report will be sent once this threshold is reached, even if the delay between send is not
+   * yet expired.
+   *
+   * @default 4Mb
+   */
+  maxBatchUncompressedSize?: number;
+  /**
+   * The maximum time in ms between reports.
+   * @default 20s
+   */
+  maxBatchDelay?: number;
+  /**
+   * Control if traces should be always sent.
+   * If false, the traces will be batched until a delay or size is reached.
+   * Note: This is highly not recommended in a production environment
+   *
+   * @default false
+   */
+  alwaysSend?: boolean;
+  /**
+   * Timeout in ms of a trace export tentative
+   * @default 30s
+   */
+  exportTimeout?: number;
+  /**
+   * The class to be used to keep track of traces and send them to the GraphOS endpoint
+   * Note: This option is aimed to be used for testing purposes
+   */
+  reporter?: (
+    options: ApolloUsageReportOptions,
+    yoga: YogaServer<Record<string, unknown>, Record<string, unknown>>,
+    logger: YogaLogger,
+  ) => Reporter;
+  /**
+   * Called when all retry attempts to send a report to GraphOS endpoint failed.
+   * By default, the error is logged.
+   */
+  onError?: (err: Error) => void;
+  /**
+   * If false, unexecutable operation (with parsing or validation error) will not be sent
+   * @default true
+   */
+  sendUnexecutableOperationDocuments?: boolean;
```

Comment:
@EmrysMyrddin Thanks for adding this option. 👍 However, in my opinion it would be better to set the default to `false` so that it doesn't unnecessarily/unexpectedly consume the limits on the number of metrics in Apollo Studio. ApolloServerPluginUsageReporting doesn't send it there by default either 😉